test/mpi/part/pingping.c

```diff
@@ -170,12 +171,12 @@ int main(int argc, char *argv[])
                 free(desc);
 
                 for (nmsg = 1; nmsg < maxmsg; nmsg++) {
-                    MTest_dtp_init(&recv, -1, -1, count[0]);
+                    MTest_dtp_init(&recv, -1, -1, recvcnt);
 
                     recv_test((char *) recv.buf + recv.dtp_obj.DTP_buf_offset, partitions,
                               partition_count, dtype, source, comm);
 
-                    errs += MTest_dtp_check(&recv, 0, 1, count[0], errs < 10);
+                    errs += MTest_dtp_check(&recv, 0, 1, sendcnt, errs < 10);
```

Comment:
It might be better to check the status for the received datatype and count and use that to check the buffer. That way we can confirm that the implementation is reporting the right type information *and* contents.

Reply:
I should say check for the received count in the status. The datatype is already known by the receiving process, and is used as input to `MPI_Get_count`.

Reply:
I'll add status check to other `pt2pt` dtp tests. However, did we (forum) ever discuss the case that partitioned communication having different count between send and recv?

Reply:
Oh, I didn't realize this was a partitioned test. I don't think underflow is allowed in partitioned. It's not completely clear from the text, but I think that is the intent. I'll follow up on it in the forum.

Reply:
Added the last commit to add status check.